### PR TITLE
RadioComponentController fix for Custom builds

### PR DESCRIPTION
**Pull request by mnumanuyar**
Signed-off-by: numan.uyar <numan.uyar@titra.com.tr>

related to issue: #9539

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
I assume you are doing this because you have an ArduPilot custom build?

Since the RadioComponentController implementation is common to both PX4 and ArduPilot the correct change is to move this to the FirmwarePlugin base class. Moving it to QGCApplication causes it to be compiled in even if you are not using either PX4 or ArduPilot firmwares which is incorrect.

**Comment by mnumanuyar** [comment]
You are absolutely correct! 
I think FirmwarePlugin does not have a constructor, should I create one and then us it in PX4FirmwarePlugin and APMFirmwarePlugin, or is there a better way?

**Comment by DonLakeFlyer** [comment]
Yes, create a constructor for FirmwarePlugin. I don't think you need to mod px4 or apm firmware plugin code to make it work,.

**Comment by mnumanuyar** [comment]
It should be ok now

File: src/FirmwarePlugin/FirmwarePlugin.cc [hunk 1]
```diff
@@ -48,6 +49,12 @@ FirmwarePluginFactoryRegister* FirmwarePluginFactoryRegister::instance(void)
     return _instance;
 }
 
+
+FirmwarePlugin::FirmwarePlugin(void)
+{
+    qmlRegisterType<RadioComponentController>       ("QGroundControl.Controllers",                       1, 0, "RadioComponentController");
```
Comment (DonLakeFlyer) [inline src/FirmwarePlugin/FirmwarePlugin.cc:55]: Can you pull the tabs out of here?

File: src/FirmwarePlugin/FirmwarePlugin.h [hunk 2]
```diff
@@ -41,6 +41,8 @@ class FirmwarePlugin : public QObject
     Q_OBJECT
 
 public:
+    FirmwarePlugin ();
```
Comment (DonLakeFlyer) [inline src/FirmwarePlugin/FirmwarePlugin.h:44]: Also the extra space here to match coding style

**Review by DonLakeFlyer — CHANGES_REQUESTED** [review]
Should be all god after the minor spacing fixes

**Comment by DonLakeFlyer** [comment]
@mnumanuyar Can you update?

**Comment by DonLakeFlyer** [comment]
@mnumanuyar Can you update?

**Comment by mnumanuyar** [comment]
sorry, I posted somewhere else but forget to put it also here. Unfortunately I am a little busy with work related stuff. I will probably have more time in 2-3 weeks.

**Comment by mnumanuyar** [comment]
Also thank you for merging and care

**Pull request merged**